src/adams_methods.jl

```diff
@@ -0,0 +1,655 @@
+            # Error Check for step size `h` and Variable next step selection:
+            #     Check error to determine if step size should be accepted:
+            #     ->if yes, compute `next_h` for next step
+            #     ->if no, use this `next_h`` for next attempt at this step
+            #---################################################################
+
+            #Don't use adaptive steps until n >= steporder + 2
+            if n==1
+                #TODO: adaptive step selection for n<steporder +2
+                successful_step = true
+                next_h = h/4
+            elseif n<steporder+2
+                yerr_k = h*(g[steporder+1] - g[steporder])*ϕ[3,steporder+1]
+                err_k, next_h,timeout_VS = stepsize_hw92!(h, tdir, y[n], y_np1, yerr_k, steporder,
+                                timeout_VS, dof, abstol, reltol, maxstep, norm)
+                if err_k <= 1
+                    successful_step = true
+                else
+                    successful_step = false
+                    h = fac*next_h # make sure fac < 1
+                end
+            else
+                # We calculate local truncation error, LE_{k+1}; the difference
+                # between implicit results for order k and k+1.
+
+                # We need ϕ_{k+1}(n+1) and g_{k+1}(n);the latter has already
+                # been calculated above, the former still needs calculation
+                ϕ_coefs!(ϕ, ϕstar,dy_np1,steporder+2)
+
+                # Calculate ||LE_{k+1}|| and next stepsize
+                yerr_kp1 = h*(g[steporder+2] - g[steporder+1])*ϕ[3,steporder+2]
+                err_kp1, next_h, timeout_VS = stepsize_hw92!(h,tdir,y[n],y_np1,
+                                            yerr_kp1, steporder, timeout_VS,dof,
+                                            abstol, reltol, maxstep, norm)
+
+                # Harier equation 7.4: If ||LE_{k+1}|| =< 1
+                # then step size is accepted and use next_h for next step
+                # Else, then step size is rejected, reattempt with fac*next_h
+                if err_kp1 <= 1
+                    successful_step = true
+                else
+                    successful_step = false
+                    h = fac*next_h # make sure fac < 1
+                end
+            end # End of error check for current step size h
+        end # End of loop for finding successfull step size for nth step
+            # repeats if success_step == false
+
+
+
+        #--#####################################################################
+        # Interpolation for dense output
+        #--#####################################################################
+        # interpolate specified points
+        while iter <= length(tspan) && tdir*tspan[iter]<tdir*t_np1
+            h_iter = tspan[iter]-t[n]
+
+            # Calculation of g_{j}(n)coefficients for j=0,...,k+1;
+            g_coefs!(g,c,h,t,tspan[iter],n,min(n,steporder+1))
```

Comment:
So this is exactly what I do. I am evaluating the integral of the polynomial from t_n to each different tspan point between t_n and t_{n+1}. This, however, requires calculating new g(n) coefficients. See the following formula from Harier vol 1, page 398
![integration](https://cloud.githubusercontent.com/assets/12954944/16810226/264a67b0-48f1-11e6-90c9-8073c1a39475.png)